minor fix for typo, indent, link, and sample code, etc

LiYing2010 opened the pull request:
### Change List
- fixed some broken links
- changed some text to link
- fixed some indent in sample code
- changed all the `’` char to `'` char
- fixed some incorrect content

sarahhaggarty commented:
Hi @LiYing2010 ! Thanks for creating this PR. I am _almost_ finished with the review but unfortunately there are just a couple items waiting for approval from members of our developer team who are currently on vacation. I should be able to finish this in the next two weeks once they are back. Sorry for the delay and thank you for your patience! 🙏

sarahhaggarty commented on docs/topics/tour/kotlin-tour-intermediate-lambdas-receiver.md:
```diff
@@ -45,7 +45,7 @@ Consider this example that extends the `StringBuilder` class:
 fun main() {
     // Lambda expression with receiver definition
-    fun StringBuilder.appendText() { append("Hello!") }
+    val appendText: StringBuilder.() -> Unit = { append("Hello!") }
```
You raise a good point. After discussing with our developer team, I'd like to change the example to not use the `StringBuilder` class anymore. Could you revert this change for now and I'll raise a separate PR to fix the example.

LiYing2010 replied:
fixed

LiYing2010 commented on docs/topics/tour/kotlin-tour-intermediate-classes-interfaces.md:
```diff
@@ -363,7 +363,7 @@ basic features but also unique behaviors. In the code sample below, complete the
 
 Then, create another child class called `SmartThermostat` that inherits from the `SmartDevice` class and implements 
-`turnOn()` and `turnOff()` functions that return print statements describing which thermostat is heating or turned off.
+`turnOn()` and `turnOff()` functions that trigger print statements describing which thermostat is heating or turned off.
```
fixed

LiYing2010 commented on docs/topics/tour/kotlin-tour-intermediate-scope-functions.md:
```diff
@@ -288,7 +288,7 @@ The example:
 * Passes a lambda expression to the `.map()` function that refers to `medals` via the `it` keyword and calls the `.uppercase()` extension function on it.
 * Uses the `.filter()` extension function on the `medals` variable.
-* Passes a lambda expression as a predicate to the `.filter()` function that refers to `medals` via the `it` keyword and checks if the length of the list contained in the `medals` variable is longer than 4 items.
+* Passes a lambda expression as a predicate to the `.filter()` function that refers to `medals` via the `it` keyword and checks if the length of the string contained in the `medals` variable is longer than 4 characters.
```
fixed

LiYing2010 commented on docs/topics/tour/kotlin-tour-intermediate-scope-functions.md:
```diff
@@ -126,7 +126,7 @@ fun main() {
 
 The example:
-* Creates a variable called `confirm`.
+* Creates a variable called `address`.
```
fixed

sarahhaggarty commented:
Thanks for updating the PR @LiYing2010 ! Can you rebase on top of master and resolve the conflicts?

LiYing2010 commented:
@sarahhaggarty 
I don't see any conflict
but I rebased it as you asked, and squashed commits, and force-pushed it, to make the commit-history clean

please check it
thank you~~

sarahhaggarty commented on docs/topics/tour/kotlin-tour-intermediate-properties.md:
```diff
@@ -204,7 +204,7 @@ In these functions:
 * The `thisRef` parameter refers to the object **containing** the delegated property. By default, the type is set to `Any?`, but you may need to declare a more specific type.
 * The `property` parameter refers to the property whose value is accessed or changed. You can use this parameter to access information
-like the property's name or type. By default, the type is set to `Any?`. You don't need to worry about changing this in your code.
+like the property's name or type. By default, the type is set to `KProperty<*>`. You don't need to worry about changing this in your code.
```
```suggestion
like the property's name or type. By default, the type is set to `KProperty<*>` but you can also use `Any?`. You don't need to worry about changing this in your code.
```

sarahhaggarty commented on docs/topics/tour/kotlin-tour-intermediate-properties.md:
```diff
@@ -230,7 +230,7 @@ want from the `get()` function of the delegated property to the `getValue()` ope
     var cachedValue: String? = null
 
-    operator fun getValue(thisRef: Any?, property: Any?): String {
+    operator fun getValue(thisRef: Any?, property: KProperty<*>): String {
```
```suggestion
    operator fun getValue(thisRef: Any?, property: Any?): String {
```

sarahhaggarty commented on docs/topics/tour/kotlin-tour-intermediate-properties.md:
```diff
@@ -253,7 +253,7 @@ Now you can delegate the property that you want to cache (`val displayName`) to
     var cachedValue: String? = null
 
-    operator fun getValue(thisRef: User, property: Any?): String {
+    operator fun getValue(thisRef: User, property: KProperty<*>): String {
```
```suggestion
    operator fun getValue(thisRef: User, property: Any?): String {
```

sarahhaggarty reviewed (changes requested):
Thanks for rebasing @LiYing2010 !
Unfortunately we can't add `KProperty<*>` in the runnable samples due to a limitation of Kotlin playground. Could you change it back to `Any?` as per my suggestions? Then we're ready to merge!

LiYing2010 commented:
@sarahhaggarty 
OK
I reverted commit https://github.com/JetBrains/kotlin-web-site/commit/8f372385fc2bca10460a5ecafe0e1407311c52c6 ,
and changed the doc content as you suggested

please review it again
thank you

sarahhaggarty commented:
Thank you for your contribution @LiYing2010 ! 🚀 And thanks again for your patience 🙏

LiYing2010 commented:
@sarahhaggarty 
thank you very much for reviewing and accepting my PR




> btw, I found something else need to fix, and I will send another PR

I will send a new PR soon for this
